### PR TITLE
More work on graph plotting

### PR DESCRIPTION
**Pull request by danielchriscarter**

### CODE REVIEW

File: builds.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+                        content = infile.readline()
+                    pprint.pprint(device.release_first_seen)
+                    _devices.append(device)
+            except EOFError:
+                print('Invalid file:', dump)
+                continue
+        save_devices_to_file()
+    return _devices
+
+def analyse_vulnerabilities(vulnerabilities, dates, string_keys=False):
```
Comment (drt24) [inline builds.py:251]: What is string_keys for?

File: builds.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+            # Store the number of devices which are of a particular status
+            number_per_status_type[status] += 1
+        if string_keys:
+            figures_per_date[str(date)] = number_per_status_type
+        else:
+            figures_per_date[date] = number_per_status_type
+    return figures_per_date
+
+def analyse_vulnerability_exploits(vulnerabilities, dates, string_keys=False):
+    """Get the number of devices by exploits possible on thos devices, on each of the dates passed"""
```
Comment (drt24) [inline builds.py:274]:
```suggestion
    """Get the number of devices by exploits possible on those devices, on each of the dates passed"""
```

**Review by drt24 — APPROVED** [review]
Looks good to me.

File: builds.py (hunk 1 above)
Comment (danielchriscarter) [reply builds.py:251]: The JSON export library doesn't allow datetimes to be dictionary keys, so there has to be an option to represent the dates as strings for JSON export

File: builds.py (hunk 2 above)
Comment (danielchriscarter) [reply builds.py:274]: Thanks for spotting - now fixed

**Pull request merged**